src/nccl_ofi_net.c

```diff
@@ -1402,26 +1409,42 @@ static ncclResult_t ofi_connect(int dev, void *handle, void **sendComm)
 	req->dev = sComm->dev;
 	req->direction = NCCL_OFI_SEND;
 
+	conn_info = (struct connection_info*)calloc(1, sizeof(*conn_info));
+	if (OFI_UNLIKELY(conn_info == NULL)) {
+		NCCL_OFI_WARN("Couldn't allocate connection_info for dev %d", dev);
+		ret = ncclSystemError;
+		goto error;
+	}
+	conn_info->ep_namelen = sizeof(conn_info->ep_name);
+
 	/* Get local EP address to transfer in the connect message */
 	ret = fi_getname(&(nccl_ofi_component[dev]->ep->fid),
-			 (void *)&local_ep_addr,
-			 &namelen);
+			 (void *)conn_info->ep_name,
+			 &(conn_info->ep_namelen));
 	if (ret != 0) {
 		NCCL_OFI_WARN("Call to fi_getname() failed with RC: %d, ERROR: %s",
 			      ret, fi_strerror(-ret));
 		ret = ncclSystemError;
 		goto error;
 	}
 
+	connect_to_self = (memcmp(conn_info->ep_name, remote_ep_name, conn_info->ep_namelen) == 0);
+
+	if (connect_to_self) {
+		conn_info->req = req;
+		/* for future freeing conn_info buffer */
+		conn_info->memory_buffer = conn_info;
+	}
```

Comment:
Branch out to another function? Branching out will help moving around the code when 2.12 support will be added.

Reply:
ok

Reply:
done